### DISCUSSION TITLE
[Feature] Upload large files via streaming instead of buffering

### DISCUSSION THREAD

**Discussion by JordanOlivet** [immich-app/immich#11076] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I recently setup Immich on my home server and it is great.

But I ran into one small issue, here is my setup.
I use a reverse proxy (traefik) to get access to my Immich instance through internet using a DNS and all my clients (actually only two, the web client and the Immich android app on my smartphone) uses the DNS url.

I ran the automatic upload with my smartphone and all my photos and videos have been uploaded but 3 videos.

I tested to upload them throught the web client but the same problem, no upload.
To go further, I installed the Immich cli to try to upload the videos with that and same, the upload failed. But I got and additional info, an error 413 entity too large.
So I search online and found that this could be caused by the reverse proxy or DNS manager (for it is cloudflare).

I configure my reverse proxy but no change but I found that Cloudflare limit to 100Mb the size of the request content, in the lowest tier (my case).
To be sure, I retested Immich cli but with my local ip, on my local network and that worked.

So, I dont want to pay a higher tier on Cloudflare (and the highest tier allows only 500Mb or you have to be an enterprise and negotiate) but I was thinking that if I reach the max request size, this is maybe because Immich send the whole file in one single time instead of streaming it to the server and perform a md5 check or else at the end to ensure that the file has been fully sent.

I think this approach would eliminate completely the problem and also maybe mitigate some problems with timeout on large files (I rapidly saw that while searching).

I dont know if this is possible or even already existing (I searched but found nothing related).
If it can resolve this problem it would be pretty useful for users (if it really bypass request max size).
For example, I found another person with similar upload problem [here](https://github.com/immich-app/immich/discussions/10331).

Thanks in advance and ask any question about it, I gladly answer it.

Immich server version : 1.108.0
Immich android app version : 1.108.0

PS : Sorry for my bad english, this is not my first language.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#1674

**Comment by JordanOlivet** [comment]
Oh I didnt find this discussion. Thanks, I be following it closely.

**Comment by Sylvain303** [comment]
Note that #1674 is now locked.

**Comment by chicuongle** [comment]
actually I found new feature of immich to automatically switch between different network fix issue related to uploading big files. As long as we at home, big files can be uploaded properly.

**Reply by Sylvain303** [reply]
what is this feature, you mentioned @chicuongle?

**Reply by chicuongle** [reply]
This one https://github.com/immich-app/immich/pull/14437

**Reply by Sylvain303** [reply]
@chicuongle  oh great.

If I understand well the linked feature, the mobile app can't still stream chunk for large content (say video) but the user can configure alternative entrypoint for upload some bigger content.

Like local area network entrypoint when connected back to the LAN where the destination server immich could be reached. 🤔

Nice trick but only available for some specific usecase.

Which could match part of my usage. But not for my users.

Thanks a lot for the link 👍️
